### PR TITLE
fix: crash on exit in `explorer-desktop`

### PR DESCRIPTION
**Pull request by Kinerius**
## What does this PR change?

When unity is being shut down, objects may be destroyed before the desired flow is completed. 
This PR Prevents multiple errors and exceptions during that phase.

Fixes: decentraland/unity-renderer#1566

## Extra

When the game is being destroyed, for some reason, we are trying to destroy `Transform` components and unity doesn't allow that, throwing an ugly error with no stack trace. So a condition was added to `Utils.SafeDestroy`

## How to test the changes?

Try Desktop build that's using this branch https://github.com/decentraland/explorer-desktop/pull/42
Exit the application and no bug report splash should be seen.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: unity-renderer/ProjectSettings/Packages/com.unity.testtools.codecoverage/Settings.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "m_Name": "Settings",
```
Comment (lorux0) [inline unity-renderer/ProjectSettings/Packages/com.unity.testtools.codecoverage/Settings.json:2]: is this file needed?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Environment.cs [hunk 2]
```diff
@@ -114,6 +115,15 @@ public void Dispose()
                 platform?.Dispose();
                 hud?.Dispose();
             }
+            public void OnDestroy()
+            {
+                IsBeingDestroyed = true;
+            }
+        }
+
+        public static void OnDestroy()
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Environment.cs:124]: you could do `Environment.i.OnDestroy()` instead, and avoid this method

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Environment.cs [hunk 3]
```diff
@@ -114,6 +115,15 @@ public void Dispose()
                 platform?.Dispose();
                 hud?.Dispose();
             }
+            public void OnDestroy()
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Environment.cs:118]: have you tried using `Dispose` method for flagging `IsBeingDestroyed` ?

**Review by lorux0 — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Environment.cs (hunk 3 above)
Comment (Kinerius) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Environment.cs:118]: I needed a way to flag this before the `Dispose` as the PluginSystem is the first one to be disposed and needs to know if everything is being destroyed

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Environment.cs [hunk 4]
```diff
@@ -93,6 +93,7 @@ public class Model
             public readonly PlatformContext platform;
             public readonly WorldRuntimeContext world;
             public readonly HUDContext hud;
+            public bool IsBeingDestroyed { get ; private set; }
```
Comment (BrianAmadori) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Environment.cs:96]: `Environment` shouldn't be storing this kind of state. `DataStore_Common` is a better alternative.

**Review by BrianAmadori — CHANGES_REQUESTED** [review]
Really glad that you fixed this 🍻

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Environment.cs (hunk 4 above)
Comment (Kinerius) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Environment.cs:96]: This is the idea I was looking for!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/Utils/Utils.cs [hunk 5]
```diff
@@ -278,6 +278,9 @@ public static GameObject AttachPlaceholderRendererGameObject(UnityEngine.Transfo
 
         public static void SafeDestroy(UnityEngine.Object obj)
         {
+            if (obj is Transform)
```
Comment (BrianAmadori) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/Utils/Utils.cs:281]: Why we'd need this condition?

**Review by BrianAmadori — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/Utils/Utils.cs (hunk 5 above)
Comment (Kinerius) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/Utils/Utils.cs:281]: When the game is being destroyed, for some reason, we are trying to destroy `Transform` components and unity doesn't allow that, throwing an ugly error with no stack trace.  
This condition avoided those error messages.
